protocol/messenger_handler.go

```diff
@@ -3008,9 +3015,14 @@ func (m *Messenger) isMessageAllowedFrom(publicKey string, chat *Chat) (bool, er
 	return contact.added(), nil
 }
 
-func (m *Messenger) updateUnviewedCounts(chat *Chat, mentionedOrReplied bool) {
-	chat.UnviewedMessagesCount++
-	if mentionedOrReplied {
+func (m *Messenger) updateUnviewedCounts(chat *Chat, message *common.Message) {
+	if chat == nil {
+		return
+	}
+	if chat.LastMessage == nil || chat.LastMessage.IsSystemMessage() || message.Timestamp > chat.LastMessage.Timestamp+IncreaseUnviewedMessagesCountTimeout {
```

Comment:
An important update here compared to the previous version:  all the (potentially important) system messages always increase counter. That's done mainly because of the bug we had in the previous PR (https://github.com/status-im/status-go/pull/3845): all the messages sent in two minutes after the contact request was accepted were not increasing the counter, now it's fixed.